src/main/java/HttpHandlerSelector.java

```diff
@@ -1,114 +1,24 @@
 package com.td.HttpServer;
-import java.util.*;
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
-import java.io.IOException;
 
-// TODO This class is too big, and has multiple responsibilities.
-//      It needs to be broken up into multiple classs.
-//      Two classes that could be pulled out of this are
-//         HttpResponseBuilder & HeaderBuilder
+import java.util.*;
 
 public class HttpHandlerSelector {
 
-  IFileIO fileIO;
-  HashMap<String, String> standardHeaders;
-
-  private final String CODE_200 = "HTTP/1.1 200 OK";
-  private final String CODE_404 = "HTTP/1.1 404 Not Found";
-  private final String CODE_400 = "HTTP/1.1 400 Bad Request";
-
-  private final String HTML = ".html";
+  HttpHandleRequest getRequest;
 
-  public HttpHandlerSelector(IFileIO fileIO) {
-    this.fileIO = fileIO;
-    standardHeaders = new HashMap<String, String>();
-    standardHeaders.put("Server", "Simple Http Server written in Java");
+  public HttpHandlerSelector(HttpHandleRequest getRequest) {
```

Comment:
So, this just made me realize this might be an awful design choice. My original thought was that has I hadn't more types of requests that the constructor would grow to include those as well. So after the POST story it would look like.
`public HttpHandlerSelector(HttpHandleRequest getRequest, HttpHandleRequest postRequest)`
I would then add a HashMap<String, object> with the method types as keys and the instances of HttpHandleRequest as values. This would then remove the `if` check on `request.method` so that it didn't turn into a giant `if`, `else if` block. But, I'm thinking maybe there is a cleaner way to do this. I might need to think on this for a moment.